src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,86 @@
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+    private final Map<Class<?>, Class<?>> searchInterfaceImplMap = new HashMap<>();
+
+    private Injector() {
+        searchInterfaceImplMap.put(FileReaderService.class, FileReaderServiceImpl.class);
+        searchInterfaceImplMap.put(ProductParser.class, ProductParserImpl.class);
+        searchInterfaceImplMap.put(ProductService.class, ProductServiceImpl.class);
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        //Create temp null Object
+        Object clazzImplInst = null;
+        //Create generic class to find Implementation in my classes
+        Class<?> clazz = findImplementationInstance(interfaceClazz);
+        //Check to - have annotations component class
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Can`t create an instance at this class!");
+        }
+
+        Field[] declaredField = clazz.getDeclaredFields();
+
+        for (Field field : declaredField) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object filedInstance = getInstance(field.getType());
+
+                clazzImplInst = createNewIstance(clazz);
+
+                field.setAccessible(true);
+                try {
+                    field.set(clazzImplInst, filedInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can`t initialize field value. Class"
+                            + clazz.getName()
+                            + " Field " + field.getName(), e);
+                }
+            }
+        }
+        if (clazzImplInst == null) {
+            clazzImplInst = createNewIstance(clazz);
+        }
+        return clazzImplInst;
+    }
+
+    private Object createNewIstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        //Create generic constructor to
+        Constructor<?> constructor;
+
+        try {
+            constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException("No such method." + e);
+        } catch (InvocationTargetException | InstantiationException | IllegalAccessException e) {
```

Comment:
checklist issue, please always read it before submitting your solution 😥
![image](https://github.com/mate-academy/jv-dependency-injection/assets/49767049/c9c8bf76-8bbc-42a3-a7f5-9ff551e7fec9)